### DISCUSSION TITLE
Running interactive apply with a pager

### DISCUSSION THREAD

**Discussion by moray95** [helmfile/helmfile#1428] [Ideas]
Hello,

I would like to run Helmfile apply with a pager, similar to running `helmfile diff | less && helmfile apply`, with an additional confirmation before the apply. This would enable to properly examine the diff especially for larger deployments where the diff might not fit into the terminal buffer.

Currently, we need to run diff + apply commands separately as `helmfile apply -i | less` does not work due to less capturing stdin, so there is no way to approve/reject the interactive apply. 

I'm not sure if such a feature already exists but couldn't find any mentions in the documentation or cli options.

**Comment by yxxhero** [comment]
sorry. I don't get your meaning. could you explain more?

**Reply by moray95** [reply]
Alright, let me try to summarize it more clearly.

Issue:
When running `helmfile apply -i` with large deployments, the diff gets bigger than the terminal buffer and I am unable to review the whole diff.

Current workaround:
First run diff with a pager, then run non-interactive apply:
```
function approve() { 
  echo -n "Approve ? ";
  read confirm && [[ $confirm == [yY] || $confirm == [yY][eE][sS] ]] || exit 1 
}
helmfile diff | less
approve && helmfile apply
```

Proposed solution:
Make it possible to show the interactive apply diff with a pager (for example `helmfile apply -i --pager=less`), this way we won't have any issues viewing the diff for large deployments.
